lib/net.js

```diff
@@ -434,6 +434,10 @@ function afterShutdown() {
 // of the other side sending a FIN.  The standard 'write after end'
 // is overly vague, and makes it seem like the user's code is to blame.
 function writeAfterFIN(chunk, encoding, cb) {
+  if (!this.writabledEnded) {
+    return stream.Duplex.prototype.write.call(this, chunk, encoding, cb);
```

Comment:
```suggestion
    return ReflectApply(stream.Duplex.prototype.write, this, [chunk, encoding, cb]);
```

Reply:
Why is this better?

Reply:
Do you mean for using `Reflect.apply` over `Function.prototype.call`? See https://github.com/nodejs/node/pull/35838#pullrequestreview-519781441, `ReflectApply` has less overhead.

Reply:
I don't see the reason to need to change code of this form to use something like `Reflect.apply()` or `FunctionPrototypeApply()`. What does it actually buy us?

Reply:
That's on the spirit of using primordials in core:

https://github.com/nodejs/node/blob/ab587ca46b49190564a4d808115af3a9b5ba71b6/lib/internal/per_context/primordials.js#L5-L7

The idea is it should be possible for users to do something like `node -e "Function.prototype.call = null;console.log(4)"` and not getting errors from the internals.